enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/context.tsx

```diff
@@ -1,83 +1,142 @@
 import {
   createContext,
-  type PropsWithChildren,
+  useCallback,
   useContext,
-  useMemo,
+  useEffect,
+  useState,
+  type PropsWithChildren,
 } from "react";
 
 import type { SdkPluginsConfig } from "embedding-sdk";
+import {
+  loadSdkQuestion,
+  type LoadSdkQuestionParams,
+} from "embedding-sdk/lib/load-question";
+import {
+  runQuestionOnQueryChangeSdk,
+  runQuestionOnNavigateSdk,
+} from "embedding-sdk/lib/run-question-query";
 import { useSdkSelector } from "embedding-sdk/store";
 import { getPlugins } from "embedding-sdk/store/selectors";
+import type {
+  NavigateToNewCardParams,
+  SdkQuestionResult,
+} from "embedding-sdk/types/question";
+import { useDispatch } from "metabase/lib/redux";
 import type { Mode } from "metabase/visualizations/click-actions/Mode";
 import { getEmbeddingMode } from "metabase/visualizations/click-actions/lib/modes";
+import type Question from "metabase-lib/v1/Question";
 
-import {
-  type UseLoadQuestionParams,
-  useInteractiveQuestionData,
-  useLoadQuestion,
-} from "./hooks";
-
-type InteractiveQuestionContextType = {
+interface InteractiveQuestionContextType extends SdkQuestionResult {
   plugins: SdkPluginsConfig | null;
   mode: Mode | null | undefined;
   isQuestionLoading: boolean;
   resetQuestion: () => void;
   onReset?: () => void;
   onNavigateBack?: () => void;
-};
+  onQuestionChange: (question: Question) => void;
+  onNavigateToNewCard: (params: NavigateToNewCardParams) => void;
+}
 
 /**
  * Note: This context should only be used as a wrapper for the InteractiveQuestionResult
  * component. The idea behind this context is to allow the InteractiveQuestionResult component
  * to use components within the ./components folder, which use the context for display
- * and functions. Any data that can be referenced from the store should be placed in
- * the useInteractiveQuestionData hook.
+ * and functions.
  * */
 export const InteractiveQuestionContext = createContext<
   InteractiveQuestionContextType | undefined
 >(undefined);
 
 type InteractiveQuestionProviderProps = PropsWithChildren<
-  UseLoadQuestionParams & {
+  {
     componentPlugins?: SdkPluginsConfig;
     onReset?: () => void;
     onNavigateBack?: () => void;
-  }
+  } & LoadSdkQuestionParams
 >;
 
 export const InteractiveQuestionProvider = ({
-  children,
   location,
   params,
   componentPlugins,
   onReset,
   onNavigateBack,
+  children,
 }: InteractiveQuestionProviderProps) => {
-  const { isQuestionLoading, resetQuestion } = useLoadQuestion({
-    location: location,
-    params: params,
-  });
+  const dispatch = useDispatch();
+
+  const [result, setQuestionResult] = useState<SdkQuestionResult>({});
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
```

Comment:
We use React states here instead of Redux as we support multiple instances of interactive questions.

Of course, this could be implemented in Redux by assigning a random identifier on mount, but I think the logic needed for SDK is fairly simple so we can rely on React Context here.